Editorial Pass on Congestion Control and Loss Detection (#178)
quicwg/ack-frequency: draft-ietf-quic-ack-frequency.md

## Congestion Control

A sender needs to be responsive to notifications of congestion, such as
a packet loss or an ECN CE marking. Decreasing the acknowledgement frequncy
can delay a sender's response to network congestion or cause it to underutilize
the available bandwidth.

To limit the consequences of reduced acknowledgement frequency, a sender
SHOULD cause a receiver to send an acknowledgement at least once per RTT if
there are unacknowledged ack-eliciting packets in flight.

A sender can accomplish this by setting the Ack-Eliciting Threshold
to a value no larger than the current congestion window or the Request Max Ack
Delay value to no more than the estimated round trip. Alternatively,
a sender can accomplish this by sending an IMMEDIATE_ACK frame once per
round trip, though if the packet containing an IMMEDIATE_ACK is lost,
detection of that loss will be delayed by the reordering threshold or requested
max ack delay.

> Note that the congestion window and the RTT are dynamic and therefore might require

Author: In RFC9002, we don't use 'size of the congestion window', and instead use 'congestion window', so I'd like to stick with that for this draft.

sending frequent ACK_FREQUENCY frames to ensure optimal performance.

It is possible that the RTT is smaller than the receiver's timer granularity,
as communicated via the 'min_ack_delay' transport parameter, preventing the
receiver from sending an acknowledgment every RTT in time unless packets are
acknowledged immediately.  Additionally, Reordering Threshold values other than 1

> can be harmful, because it can delay time threshold loss detection.

Reviewer: Honestly I do not understand what the last sentence is trying to argue, regardless of the mechanism being packet number threshold or time threshold. It does not seem to me that it is specific to the condition being mentioned at the beginning of this paragraph.

That said, I understand that this is a rewrite of an existing sentence and that this PR is meant to be editorial.

> can be harmful, because it can delay time threshold loss detection.

Author: The idea is that if min_ack_delay is larger than RTT, it'll can more than an RTT to receive a delayed ACK.

I updated the text to clarify what the purpose of the sentence is.

> can be harmful, because it can delay time threshold loss detection.

Reviewer: Thank you. I think what the text says makes sense.

> can be harmful, because it can delay time threshold loss detection.

Author: ```suggestion
can be harmful, because it can delay loss detection more than an RTT.
```
